lib/daru/io/importers/importer.rb

```diff
@@ -0,0 +1,38 @@
+module Daru
+  module IO
+    module Importers
+      class Importer
+        def self.guess_parse(keys, vals)
+          case vals.first
+          when Array
+            case vals.first.first
+            when Hash
+              # Array of hashes
+              # key a :
+              # [
+              #   { x: 1, y: 2 },
+              #   { x: 3, y: 4 }
+              # ]
+              Daru::DataFrame.new vals.flatten
+            else
+              # Hash containing Array
+              # key a :
+              # {
+              #   x: [1,2,3,4]
+              #   y: [5,6,7,8]
+              # }
+              Daru::DataFrame.rows vals.transpose, order: keys
+            end
+          when Hash
+            # Array containing Hash
+            # [
+            #   key a: { x: 1, y: 2 }
+            #   key b: { x: 3, y: 4 }
+            # ]
+            Daru::DataFrame.new vals.flatten, index: keys
+          end
+        end
```

Comment:
Maybe this way it will look cleaner (each condition is a bit more complicated, but structure is pretty visible):
```ruby
test = vals.first
case
when test.is_a?(Array) && test.first.is_a?(Array)
when test.is_a?(Array) && test.first.is_a?(Hash)
when test.is_a?(Hash)
```
...but I am not insisting on this approach, if you don't find it readable.